.github/workflows/ci.yaml

```diff
@@ -2,43 +2,40 @@ name: CI
 on:
   pull_request:
   workflow_dispatch:
-  push:
-    branches:
-      - main
 jobs:
   build:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        node-version: [14.15.1]
     steps:
     - uses: actions/checkout@v2
-    - name: Use Node.js ${{ matrix.node-version }}
+    - name: Read .nvmrc
+        run: echo ::set-output name=NVMRC::$(cat .nvmrc)
+        id: nvm
+    - name: Use Node.js ${{ steps.nvm.outputs.NVMRC }}
       uses: actions/setup-node@v2.1.4
       with:
-        node-version: ${{ matrix.node-version }}
+        node-version: "${{ steps.nvm.outputs.NVMRC }}"
     - run: ./script/build
   lint:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        node-version: [14.15.1]
     steps:
     - uses: actions/checkout@v2
-    - name: Use Node.js ${{ matrix.node-version }}
+    - name: Read .nvmrc
+        run: echo ::set-output name=NVMRC::$(cat .nvmrc)
+        id: nvm
+    - name: Use Node.js ${{ steps.nvm.outputs.NVMRC }}
       uses: actions/setup-node@v2.1.4
       with:
-        node-version: ${{ matrix.node-version }}
+        node-version: "${{ steps.nvm.outputs.NVMRC }}"
     - run: ./script/lint
   test:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        node-version: [14.15.1]
     steps:
     - uses: actions/checkout@v2
-    - name: Use Node.js ${{ matrix.node-version }}
+    - name: Read .nvmrc
```

Comment:
It's a shame we have to repeat this code in all the jobs - hopefully GitHub will provide an easier way to define shared/reusable steps soon!

Reply:
Yeah, I had a look into how we could wrap this up into one step and couldn't find a simple solution for now but hopefully it's something that will become easier to solve in the future. An alternative in the meantime would be to run a `ci` script which ran all of these commands so that we only required one check.